pkg/apiserver/proxy_test.go

```diff
@@ -367,3 +367,58 @@ func TestProxyUpgrade(t *testing.T) {
 		t.Fatalf("expected '%#v', got '%#v'", e, a)
 	}
 }
+
+func TestRedirectOnMissingTrailingSlash(t *testing.T) {
+	table := []struct {
+		// The requested path
+		path string
+		// The path requested on the proxy server.
+		proxyServerPath string
+	}{
+		{"/trailing/slash/", "/trailing/slash/"},
+		{"/", "/"},
+		// "/" should be added at the end.
+		{"", "/"},
+		// "/" should not be added at a non-root path.
+		{"/some/path", "/some/path"},
+	}
+
+	for _, item := range table {
+		proxyServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+			if req.URL.Path != item.proxyServerPath {
+				t.Errorf("Unexpected request on path: %s, expected path: %s, item: %v", req.URL.Path, item.proxyServerPath, item)
+			}
+		}))
+		defer proxyServer.Close()
+
+		serverURL, _ := url.Parse(proxyServer.URL)
+		simpleStorage := &SimpleRESTStorage{
+			errors:                    map[string]error{},
+			resourceLocation:          serverURL,
+			expectedResourceNamespace: "ns",
+		}
+
+		handler := handleNamespaced(map[string]rest.Storage{"foo": simpleStorage})
+		server := httptest.NewServer(handler)
+		defer server.Close()
+
+		proxyTestPattern := "/api/version/proxy/namespaces/ns/foo/id" + item.path
+		req, err := http.NewRequest(
+			"GET",
+			server.URL+proxyTestPattern,
+			strings.NewReader(""),
+		)
+		if err != nil {
+			t.Errorf("unexpected error %v", err)
+			continue
+		}
+		resp, err := http.DefaultClient.Do(req)
```

Comment:
Can you add a comment here to the effect that the default client follows the redirect, which is why this test works? Took me a minute to figure it out. :)